Add --run-slow flag to runtests.py

dwoz opened the pull request:
### What does this PR do?

Add --run-slow flag to runtests.py

### Commits signed with GPG?
Yes

dwoz committed:
Handle un-ordered test decorators

s0undt3ch commented on tests/support/helpers.py:
```diff
@@ -153,10 +153,51 @@ def setUp(self, *args, **kwargs):
     # We're simply decorating functions
     @functools.wraps(caller)
-    def wrap(cls):
+    def wrap(cls, *args, **kwargs):
```
Added `*args, **kwargs` because of `require_system_grains` usage?

dwoz replied:
yes, if multiple decorators are used this would fail depending on the order. For example

```
@with_tempfile
@slowTest
def foo(): pass
```

Would fail because the method returned by `slowTest` isn't expecting any arguments or keywords. This change resolves that problem for slowTest, destrictiveTest, and expensiveTest.

s0undt3ch commented on tests/support/parser/__init__.py:
```diff
@@ -635,6 +638,11 @@ def _validate_options(self):
 
+        if not os.environ.get("SLOW_TESTS", None):
+            # Set the required environment variable in order to know if
+            # destructive tests should be executed or not.
```
`s/destructive/slow`

s0undt3ch reviewed (dismissed):
The issue I found was in a comment, not a blocker to merge

s0undt3ch commented:
Also, you have conflicts to resolve now

waynew commented on tests/unit/fileserver/test_gitfs.py:
```diff
@@ -252,8 +246,8 @@ class GitFSTestFuncs(object):
-    @skipIf(True, "SLOWTEST skip")
-    @skipIf(True, "SLOWTEST skip")
+    @slowTest
+    @slowTest
```
I guess I really wanted to skip this and the next one, too :joy:

dwoz replied:
Fixed these in several files that had them.

dwoz committed:
Fix wart in docker tests with slowTest decorator

waynew reviewed (changes requested):
Uh... did the decorators disappear? :confused:

s0undt3ch commented:
Black is complaining

s0undt3ch commented on tests/unit/test_doc.py:
```diff
@@ -0,0 +1,597 @@
+# -*- coding: utf-8 -*-
```
This file was deleted. Let's not add it back.

dwoz committed:
Fix runner tests on windows

waynew reviewed (changes requested):
looks like https://jenkinsci.saltstack.com/job/pr-amazon2-py3/job/PR-56872/17/testReport/junit/integration.modules.test_cmdmod/CMDModuleTest/test_windows_powershell_script_args/ is failing but... why is it even running on not-windows?

dwoz commented:
> looks like https://jenkinsci.saltstack.com/job/pr-amazon2-py3/job/PR-56872/17/testReport/junit/integration.modules.test_cmdmod/CMDModuleTest/test_windows_powershell_script_args/ is failing but... why is it even running on not-windows?

There was a wart in the patch that started running slow tests so it's a nice little preview to what will fail when we run the full test suite again.

dwoz commented:
re-run full all

bryceml commented:
re-run all full

waynew commented:
Still appears to be failing tests?

waynew committed:
Add missing windows skip

waynew committed:
Inline our code with what `unittest.case.skip` does

s0undt3ch committed:
Inspect test classes to avoid running `setUpClass` and `tearDownClass`

If a test class has all it's tests marked to be skipped, we mark the
class to be skipped so that `setUpClass` and `tearDownClass` doesn't
even run.

s0undt3ch committed:
Speed up our skip helpers

waynew reviewed (dismissed):
*slaps hood*

this baby can fit so many squash-merges in it :+1:

Thanks @s0undt3ch for getting the time issues sorted!

s0undt3ch commented:
Bah! Need to rebase again...

dwoz commented:
looks good, we just need to resolve the conflicts now :)

s0undt3ch commented:
I'm on it

s0undt3ch commented:
Conflicts solved

s0undt3ch committed:
Add --run-slow flag to runtests.py
